### PR TITLE
Created Coding Style Guideline for Xamarin.

### PR DESCRIPTION
**Pull request by rodrigoareis**

### CODE REVIEW

File: GUIDELINES.md [hunk 1]
```diff
@@ -1,3 +1,694 @@
 # CA Technologies C# Style Guide
 
-## Please add content
\ No newline at end of file
+This document describes the C# coding style of the Xamarin Mobile SDK team. This guideline is recommended to comply with all C# implementations of our products.
```
Comment (bmckercher) [inline GUIDELINES.md]: This document describes the C# coding style of Mobile SDK for Xamarin team.

File: GUIDELINES.md [hunk 2]
```diff
@@ -1,3 +1,694 @@
...
+
+## Review Microsoft Official Coding Guideline
+
+Beyond the guidelines defined in this document, we also recommend reviewing the [Microsoft C# Guide](https://docs.microsoft.com/en-us/dotnet/csharp/index). 
+
+### Mobile SDK
+
+For more information, see [developer website](http://mas.ca.com).
+
+The Xamarin Mobile SDK consists of multiple functional products separated into individual frameworks.
```
Comment (bmckercher) [inline GUIDELINES.md]: The Mobile SDK for Xamarin

File: GUIDELINES.md [hunk 3]
```diff
@@ -1,3 +1,694 @@
...
+* [File Structure](#file-structure)
+* [Copyright Statement](#copyright-statements)
+* [Error Handling](#error-handling)
+
+
+## Documentation
+
+All code **should be properly documented** in `.cs` files. Code documentation should adhere to Microsoft Doc's format.
+
+Microsoft provides a documentation generation system based on XML comments. These comments are formally single line C♯ comments (indicated by triple slashes) containing XML tags. All comments must be directly before the code block to which the comments refer, for example:
```
Comment (bmckercher) [inline GUIDELINES.md]: All comments must be placed directly before the code block to which the comments refer, for example:

File: GUIDELINES.md [hunk 4]
```diff
@@ -1,3 +1,694 @@
...
+
+```c#
+/// <summary>  
+///  This is the class definition of CustomClass.  
+/// </summary>  
+public class CustomClass{} 
+```
+A list of recommended tags for documentation comments can be found in [Microsoft C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/recommended-tags-for-documentation-comments).
+
+Visual Studio produces compiler-generated XML documents by taking the XML comments from the code and building them into an XML file. This XML file can then be used to generate human-readable documentation in a variety of forms including HTML, MSDN style documentation and Intellisense within the code window.
```
Comment (bmckercher) [inline GUIDELINES.md]: IntelliSense

File: GUIDELINES.md [hunk 5]
```diff
@@ -1,3 +1,694 @@
...
+
+Our team uses [Monodoc](http://http://www.mono-project.com/docs/tools+libraries/tools/monodoc/) as a tool to build HTML and Intellisense documentation based on the compiler-generated XML files.
```
Comment (bmckercher) [inline GUIDELINES.md]: IntelliSense

File: GUIDELINES.md [hunk 6]
```diff
@@ -1,3 +1,694 @@
...
+* Class description
+* Property
+* Enum type
+* Constant
+* Method
+
+
+## Brace Style
+
+All braces get their own line as it is a C# convention:
```
Comment (bmckercher) [inline GUIDELINES.md]: All braces get their own line (a C# convention):

File: GUIDELINES.md [hunk 7]
```diff
@@ -1,3 +1,694 @@
...
+		    // ...
+		}
+    }
+}
+```
+
+
+## New Line
+
+Blank lines improve readability. They set off blocks of code which are in themselves logically related.
```
Comment (bmckercher) [inline GUIDELINES.md]: They set off blocks of code which are logically related.

File: GUIDELINES.md [hunk 8]
```diff
@@ -1,3 +1,694 @@
...
+* Class and interface definitions
+
+One line should always be used between:
+
+*  Methods
+*  Properties
+*  Local variables in a method and its first statement
+*  Logical sections inside a method to improve readability
+
+Note that blank lines must be indented as they would contain a statement this makes insertion in these lines much easier.
```
Comment (bmckercher) [inline GUIDELINES.md]: Indent blank lines to make it easier to insert a statement.

File: GUIDELINES.md [hunk 9]
```diff
@@ -1,3 +1,694 @@
...
+When an expression will not fit on a single line, break it up according to these general principles:
+
+* Break after a comma
+* Break after an operator
+* Prefer higher-level breaks to lower-level breaks
+* Indentation for line wraps should use 4 spaces
+
+### Long Argument Lists
+
+When your argument list grows too long, split your method invocation across multiple lines, with the first argument on a new line after the opening parenthesis of the method invocation, the closing parenthesis of the invocation on its own line at the same indentation level as the line with the opening parenthesis.
```
Comment (bmckercher) [inline GUIDELINES.md]: When your argument list grows too long, split your method invocation across multiple lines. With the first argument on a new line after the opening parenthesis of the method invocation, and the closing parenthesis of the invocation on its own line at the same indentation level as the line with the opening parenthesis.

File: GUIDELINES.md [hunk 10]
```diff
@@ -1,3 +1,694 @@
...
+* Indentation should be done using spaces — never tabs
+
+### Vertical Spacing
+
+There should be exactly one blank line between methods to aid in visual clarity and organization. Whitespace within methods should separate functionality, but having too many sections in a method often means you should refactor into several methods.
+
+
+### Blocks
+
+Indentation for blocks uses 4 spaces for optimal readability.
```
Comment (bmckercher) [inline GUIDELINES.md]: Indent blocks with 4 spaces for optimal readability

File: GUIDELINES.md [hunk 11]
```diff
@@ -1,3 +1,694 @@
...
+
+### Organization of *#region*
+
+- Order of the `#region` should be identical in both interface and class files.
+- **All of the methods** under `#region` should be in **alphabetical order**.
+
+## *using* directive
+
+* `using` directives may or may not be in alphabetical order
+* Always roup using directives by common prefix, with shorter namespaces coming before longer ones
```
Comment (bmckercher) [inline GUIDELINES.md]: Always group

File: GUIDELINES.md [hunk 12]
```diff
@@ -1,3 +1,694 @@
...
+* Preffered, namespaces should be ordered in increasing order of platform specificity, with .NET namespaces first, followed by library or component namespaces, then Xamarin namespaces, finally application namespaces
```
Comment (bmckercher) [inline GUIDELINES.md]: Prefered

File: GUIDELINES.md [hunk 13]
```diff
@@ -1,3 +1,694 @@
...
+using SomeLib;
+```
+
+## Nomenclature
+
+For naming, you should follow [Microsoft Naming Guidelines](http://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines).
+
+### Special cases
+
+Despite adopting Microsoft naming conventions as general code development guideline, there some special cases that you are required to some rules in order to be more consistent on the framework’s usability.
```
Comment (bmckercher) [inline GUIDELINES.md]: Beyond the Microsoft naming conventions for general code development, there are special rules you should follow to ensure consistency and usability of the framework.

File: GUIDELINES.md [hunk 14]
```diff
@@ -1,3 +1,694 @@
...
+//
+```
+
+**Not:**
+
+```c#
+//Don't do in-line comment like this
+```
+
+#### Multiline comments
```
Comment (bmckercher) [inline GUIDELINES.md]: Multi-line

File: GUIDELINES.md [hunk 15]
```diff
@@ -1,3 +1,694 @@
...
+
+Long comments tend to grow from smaller ones, so it's simpler to always use `//` than to switch to `/* ... */` when a comment becomes "long".
+
+**For example:**
+
+```c#
+//
+// Programming style is a set of rules or guidelines used when writing the source code for a 
+// computer program.It is often claimed that following a particular programming style will
```
Comment (bmckercher) [inline GUIDELINES.md]: program. It

**Review by bmckercher — DISMISSED** [review]
Take or leave edits as you wish.

**Review by BobbyWeber — APPROVED** [review]
Approved

**Pull request merged**